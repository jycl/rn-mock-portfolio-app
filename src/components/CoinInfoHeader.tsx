import React, { useState } from 'react';
import { View, Text, Image, StyleSheet, Pressable } from 'react-native';
import Assets from '../assets';

type PropsType = {
  onCoinHeaderCb?: (asc: boolean) => void;
};

/**
 * CoinInfoHeader component at the top of list cryptocurrency price.
 * Users can press the coin subheading to sort by name (asc, desc)
 * in a callback passed into onCoinHeaderCb.
 */
export default function CoinInfoHeader({ onCoinHeaderCb }: PropsType) {
  const [coinOrderAsc, setCoinOrderAsc] = useState(true);
  const onPressCoin = () => {
    onCoinHeaderCb && onCoinHeaderCb(!coinOrderAsc);
    setCoinOrderAsc((prev) => !prev);
  };
  return (
    <View style={[styles.container, styles.padding]}>
      <Pressable
        testID="testID_sort_by_name"
        style={styles.coinNameContainer}
        onPress={onPressCoin}>
        <Text style={styles.text}>Coin</Text>
        <Image
          style={[styles.arrow, coinOrderAsc && styles.inverted]}
          source={Assets.dropdown}
        />
      </Pressable>
      <View style={styles.priceContainer}>
        <Text style={styles.text}>Price</Text>
      </View>
      <Pressable style={styles.valueContainer}>
        <Text style={styles.text}>Value</Text>
        <Image style={styles.icon} source={Assets.info} />
      </Pressable>
    </View>
  );
}

const styles = StyleSheet.create({
  padding: {
    paddingVertical: 8,
  },
  inverted: {
    transform: [{ scaleY: -1 }],
  },
  icon: {
    width: 20,
    height: 20,
    tintColor: '#5C5C5C',
    marginLeft: 8,
  },
  valueContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  priceContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-end',
    paddingRight: 16,
  },
  coinNameContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  container: {
    flexDirection: 'row',
    backgroundColor: '#212121',
  },
  text: {
    color: '#808080',
    fontSize: 16,
  },
  arrow: {
    tintColor: 'grey',
    width: 24,
    height: 24,
  },
});
