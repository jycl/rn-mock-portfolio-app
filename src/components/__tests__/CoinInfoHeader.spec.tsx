import React from 'react';
import { create } from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';
import CoinInfoHeader from '../CoinInfoHeader';

describe('CoinInfoHeader component', () => {
  it('should match snapshot render with defined props', () => {
    const headerRender = create(<CoinInfoHeader />);
    expect(headerRender.toJSON()).toMatchSnapshot();
  });

  it('should invoke callback on press coin sort button', () => {
    const spy = jest.fn();
    const { getByTestId } = render(<CoinInfoHeader onCoinHeaderCb={spy} />);
    fireEvent.press(getByTestId('testID_sort_by_name'));
    expect(spy).toBeCalledTimes(1);
  });
});
