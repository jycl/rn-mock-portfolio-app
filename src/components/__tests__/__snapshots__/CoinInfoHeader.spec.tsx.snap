// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinInfoHeader component should match snapshot render with defined props 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#212121",
        "flexDirection": "row",
      },
      Object {
        "paddingVertical": 8,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
    testID="testID_sort_by_name"
  >
    <Text
      style={
        Object {
          "color": "#808080",
          "fontSize": 16,
        }
      }
    >
      Coin
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/dropdown.png",
        }
      }
      style={
        Array [
          Object {
            "height": 24,
            "tintColor": "grey",
            "width": 24,
          },
          Object {
            "transform": Array [
              Object {
                "scaleY": -1,
              },
            ],
          },
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
        "justifyContent": "center",
        "paddingRight": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#808080",
          "fontSize": 16,
        }
      }
    >
      Price
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#808080",
          "fontSize": 16,
        }
      }
    >
      Value
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/info.png",
        }
      }
      style={
        Object {
          "height": 20,
          "marginLeft": 8,
          "tintColor": "#5C5C5C",
          "width": 20,
        }
      }
    />
  </View>
</View>
`;
